### PR TITLE
Add Unity IAP flavor

### PR DESCRIPTION
**Pull request by vegaro**
Equivalent of https://github.com/RevenueCat/purchases-android/pull/444/

Creates a new flavor that depends on the Unity IAP compatible versions of purchases-android. Also migrated the deploy job to fastlane.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -66,9 +86,14 @@ dependencies {
...
     implementation 'androidx.annotation:annotation:1.3.0'
 
-    api("com.revenuecat.purchases:purchases:$purchases_version")
-    api("com.revenuecat.purchases:purchases-core-common:$purchases_version")
-    api("com.revenuecat.purchases:purchases-core-utils:$purchases_version")
+    latestDependenciesApi "com.revenuecat.purchases:purchases:$purchases_version"
+    latestDependenciesApi "com.revenuecat.purchases:purchases-core-common:$purchases_version"
+    latestDependenciesApi "com.revenuecat.purchases:purchases-core-utils:$purchases_version"
+    
+    // Uncomment once we upgrade to 5.0.0
```
Comment (vegaro) [inline android/build.gradle:93]: This won't work until we upgrade the dependency to 5.0.0, which is in progress in another PR.

File: .circleci/config.yml [hunk 2]
```diff
@@ -107,43 +117,14 @@ jobs:
...
-              ./gradlew closeAndReleaseRepository\
-              -Psigning.keyId=$GPG_SIGNING_KEY_ID \
-              -Psigning.password=$GPG_SIGNING_KEY_PW \
-              -Psigning.secretKeyRingFile=./secring.gpg \
-              -PmavenCentralUsername=$SONATYPE_NEXUS_TOKEN_USERNAME \
-              -PmavenCentralPassword=$SONATYPE_NEXUS_TOKEN_PASSWORD \
-              -PRELEASE_SIGNING_ENABLED=true
-              echo "Snapshot deployed!"
-            fi
-          working_directory: android
```
Comment (aboedo) [inline .circleci/config.yml:146]: 👏

File: android/build.gradle [hunk 3]
```diff
@@ -25,8 +25,19 @@ buildscript {
 apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 apply plugin: "de.mannodermaus.android-junit5"
+
+def artifactId = project.property("POM_ARTIFACT_ID")
+def publishVariant = project.property("PUBLISH_VARIANT")
+if (publishVariant == "unityIAPRelease") {
+    project.ext.POM_ARTIFACT_ID = artifactId + "-unityiap"
```
Comment (aboedo) [inline android/build.gradle:32]: 👍

File: fastlane/Fastfile [hunk 4]
```diff
@@ -82,6 +82,53 @@ platform :android do
...
+        "androidSourcesJar", "androidJavadocJar", "publish --no-daemon --no-parallel"
+      ],
+      properties: gradleProperties,
+      project_dir: 'android'
+    )
+
+    UI.verbose("Creating special version for Unity IAP with BillingClient 3 and Amazon 2 for version: #{version}")
+    gradleProperties["PUBLISH_VARIANT"] = "unityIAPRelease"
+    
+    UI.verbose("Adding -unityiap to artifact ids")
```
Comment (aboedo) [inline fastlane/Fastfile:111]: love the logging

File: fastlane/Fastfile [hunk 5]
```diff
@@ -82,6 +82,53 @@ platform :android do
...
+    
+    unless is_snapshot_version?(version)
+      gradle(
+        tasks: [
+          "closeAndReleaseRepository"
+        ],
+        properties: gradleProperties,
+        project_dir: 'android'
+      )
+      github_release(version: version)
```
Comment (aboedo) [inline fastlane/Fastfile:129]: ❤️

**Review by aboedo — APPROVED** [review]

**Comment by vegaro** [comment]
I made the flavor to depend on the same dependency so we can merge this. I will update it when upgrading to 5.0.0

**Pull request merged**